serviceclient.go

```diff
@@ -105,10 +106,10 @@ func (sc *ServiceClient) Call(req interface{}, res interface{}) error {
 // CallContext sends a request to a service provider and reads a response.
 // It allows to set a context that can be used to terminate the function.
 func (sc *ServiceClient) CallContext(ctx context.Context, req interface{}, res interface{}) error {
-	if reflect.TypeOf(req) != reflect.PtrTo(reflect.TypeOf(sc.srvReq)) {
+	if !haveSameMD5(reflect.ValueOf(req).Elem().Interface(), sc.srvReq) {
```

Comment:
`Elem()` call would panic if `req` is not a pointer, but I feel an additional check is unnecessary in that case since it would panic anyway.